lib/sentry/opentelemetry/span_storage.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Sentry.Opentelemetry.SpanStorage do
```

Comment:
This design won't work for scaling out. You will have a *single* GenServer serializing access through the whole application. You could have thousands of processes calling this at the same time, and all those requests would have to go through this process and wait for their turn. We will have to rework this pretty substantially.

For something like this, a good tool to reach for is ETS. Erlang's official OTel libraries use that as well for storing spans:

https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry/src/otel_span_ets.erl

We _might_ be able to reuse Erlang's implementation itself to store and then retrieve the spans. I'd suggest to ask around (https://opentelemetry.io/community/) or let me know if you want me to put you in touch with one of the devs of the OTel/Erlang integration—I would help but I don't have the bandwidth just this week.

Reply:
@solnic I can help/look into this today and over the weekend!

Reply:
@savhappy sounds good, thanks :)

Reply:
@whatyouhide would it be better to have two tables (one for root spans and one for children) or to store the key/value pairs accordingly to differentiate like this -> 

key for root {:root_span, span_id} 
key for children {parent_span_id, span_id}